core/src/main/java/hudson/model/User.java

```diff
@@ -142,6 +142,21 @@
      */
     public static boolean ALLOW_NON_EXISTENT_USER_TO_LOGIN = SystemProperties.getBoolean(User.class.getName() + ".allowNonExistentUserToLogin");
 
+    /**
+     * Jenkins does not allow users to be {@linkplain #impersonate(UserDetails)} impersonated} if its UserDetails
+     * return false for any of its validity attributes. These include {@link UserDetails#isEnabled()},
+     * {@link UserDetails#isAccountNonLocked()}, {@link UserDetails#isAccountNonExpired()}, and
+     * {@link UserDetails#isCredentialsNonExpired()}. These attributes are normally enforced either by the underlying
+     * {@link SecurityRealm} service or by the AuthenticationManager. Some realms return user details for these types
+     * of accounts rather than throwing an exception beforehand which are validated before the user can be
+     * impersonated. The system property {@code hudson.model.User.allowImpersonationOfDisabledUsers} can be set to
+     * {@code true} to disable this behavior.
+     *
+     * @see <a href="https://issues.jenkins-ci.org/browse/JENKINS-55813">JENKINS-55813</a>
+     */
+    @Restricted(NoExternalUse.class)
+    private static boolean ALLOW_IMPERSONATION_OF_DISABLED_USERS = SystemProperties.getBoolean(User.class.getName() + ".allowImpersonationOfDisabledUsers");
```

Comment:
Please create a downstream PR to https://www.jenkins.io/doc/book/managing/system-properties/

Reply:
This is not scalable. We need to generate that sort of documentation from sources.

Reply:
https://github.com/jenkins-infra/jenkins.io/pull/3953